resize.js

```diff
@@ -15,218 +18,216 @@ var S3 = new AWS.S3();
+var insertToken = db.prepare("INSERT INTO tokens (id, image_id, valid_until) VALUES (?,?,datetime('now','+15 minute'))");
+var consumeToken = db.prepare("DELETE FROM tokens WHERE id=? AND image_id=? AND valid_until>= datetime('now')");
 var deleteOldTokens = db.prepare("DELETE FROM tokens WHERE valid_until < datetime('now')");
 
-// Create the server
-server = http.createServer(function (req, res) {
-  if ( req.method === 'GET' ) {
-    // Get an image
-    Image.get(req, res);
-  } else if ( req.method === 'POST' && req.url === '/token' ) {
-    // Create a token for a client
-    Token.create(req, res);
-  } else if ( req.method === 'POST' ) {
-    // Process an upload
-    Image.upload(req, res);
-  } else {
-    // Error if not supported
-    log('error','405 Method not supported');
-    res.writeHead(405, 'Method not supported');
-    res.end();
-  }
-});
+// Central logging. console.log can be replaced by writing to a logfile for example
+function log(level, message) {
+    var obj = {
+        datetime: new Date(),
+        severity: level,
+        message: message
+    };
+    console.log(JSON.stringify(obj));
+}
 
-Image = {}
-Image.get = function(req, res) {
-  matches = req.url.match(/^\/(.*)_(\d+)_(\d+)(_(\d+)x)?\.(.*)/);
-  if ( matches === null ) {
-    // Invalid URL
-    log('error','404 Error for ' + res.url);
-    res.writeHead(404, 'File not found');
-    res.end();
-    return;
-  }
-  
-  fileName = matches[1];
-  resolutionX = parseInt(matches[2]);
-  resolutionY = parseInt(matches[3]);
-  if ( matches[5] !== undefined ) {
-    resolutionX *= parseInt(matches[5]);
-    resolutionY *= parseInt(matches[5]);
-  }
-  fileType = matches[6].toLowerCase();
- 
-  if ( supportedFileType(fileType) === null ) {
-    log('error','Filetype ' + fileType + ' is not supported');
-    res.writeHead(415, 'Unsupported media type');
-    res.end();
-    return;
-  }
-
-  log('info','Requesting file ' + fileName + ' in ' + fileType + ' format in a ' + resolutionX + 'x' + resolutionY + 'px resolution');
-
-  Image.checkCacheOrCreate(fileName, fileType, resolutionX, resolutionY, res);
+function supportedFileType(fileType) {
+    switch (fileType) {
+    case 'jpg':
+    case 'jpeg':
+    case 'jfif':
+    case 'jpe':
+        return 'image/jpeg';
+    case 'png':
+        return 'image/png';
+    default:
+        return null;
+    }
 }
-Image.checkCacheOrCreate = function(fileName, fileType, resolutionX, resolutionY, res) {
-    // Check if it exists in the cache
-    selectImage.get([fileName, resolutionX, resolutionY, supportedFileType(fileType)],function(err,data) { 
-      if ( !err && data ) {
-        // It is in the cache, so redirect to there
-        log('info','cache hit for ' + fileName + '.' + fileType + '(' + resolutionX + 'x' + resolutionY + 'px)');
-        res.writeHead(302, {'Location': data.url, 'Cache-Control': 'public'});
+
+var Image = {};
```

Comment:
In the JS world it is convention to only capitalize constructor functions. Since this is not a constructor function but just a regular object I'd lowercase it.